pacman/model/routing_table_by_partition/multicast_routing_table_by_partition_entry.py

```diff
@@ -13,47 +13,33 @@
 # limitations under the License.
 from __future__ import annotations
 import logging
-from typing import FrozenSet, Iterable, Optional, Union
+from typing import Iterable, Optional, Union
 from spinn_utilities.log import FormatAdapter
+from spinn_machine.base_multicast_routing_entry import (
+    BaseMulticastRoutingEntry)
 from pacman.exceptions import (
     PacmanConfigurationException, PacmanInvalidParameterException)
 
 log = FormatAdapter(logging.getLogger(__name__))
 
-_INCOMING_LINK_MASK = 0x07000000
-_INCOMING_LINK_SHIFT = 24
-_INCOMING_PROC_MASK = 0xF8000000
-_INCOMING_PROC_SHIFT = 27
-_OUTGOING_LINKS_MASK = 0x0000003F
-_OUTGOING_LINK_1 = 0x00000001
-_OUTGOING_PROCS_MASK = 0x00FFFFC0
-_OUTGOING_PROC_1 = 0x00000040
-_SPINNAKER_ROUTE_MASK = _OUTGOING_LINKS_MASK | _OUTGOING_PROCS_MASK
-_COMPARE_MASK = _INCOMING_LINK_MASK | _SPINNAKER_ROUTE_MASK
-_N_PROCS = 18
-_N_LINKS = 6
 
-
-class MulticastRoutingTableByPartitionEntry(object):
+class MulticastRoutingTableByPartitionEntry(BaseMulticastRoutingEntry):
     """
     An entry in a path of a multicast route.
     """
 
-    __slots__ = (
-        # pylint: disable=wrong-spelling-in-comment
-        # Entry made up of bits as follows:
-        # | IL = 6 bits | IP = 1 bit | OL = 6 bits | OP = 18 bits |
-        # IL = incoming link id
-        # IP = whether the source is a processor or not
-        # OL = outgoing links
-        # OP = outgoing processors
-        "_links_and_procs", )
+    __slots__ = ["_incoming_processor", "_incoming_link"]
 
     def __init__(self, out_going_links: Union[int, Iterable[int], None],
                  outgoing_processors: Union[int, Iterable[int], None],
                  incoming_processor: Optional[int] = None,
-                 incoming_link: Optional[int] = None):
+                 incoming_link: Optional[int] = None,
+                 spinnaker_route: Optional[int] = None):
```

Comment:
Do we actually make use of this here? I would have thought it would never be called in this way...

Reply:
This might be another case where we could have a "from_processor_and_links" function or something like that.  It might alternatively then hold a SpiNNMachine routing entry rather than extend it, but not sure if that would help things here...

Reply:
The main need for these is in the Mundy routers.

Allow these to be lost and I think the secondary usage of seeing is a route is defaulable can be handled at init time.